client/app/hosting/framework/envvar/hosting-framework-envvar.service.js

```diff
@@ -0,0 +1,134 @@
+angular.module("services").service(
+    "HostingFrameworkEnvvar",
+    class HostingFrameworkEnvvar {
+
+        /**
+         * @constructs HostingFrameworkEnvvar
+         * @param $q
+         * @param OvhHttp
+         * @param Hosting
+         */
+        constructor ($q, OvhHttp, Hosting) {
+            this.$q = $q;
+            this.OvhHttp = OvhHttp;
+            this.Hosting = Hosting;
+        }
+
+        /**
+         * Get list of envvars on hosting
+         * @param serviceName
+         * @param filters
+         */
+        list (serviceName, filters) {
+            const promises = [];
+
+            if (angular.isArray(filters)) {
+                filters.forEach((filter) => {
+                    promises.push(
+                        this.OvhHttp.get(`/hosting/web/${serviceName}/envVar`, {
+                            rootPath: "apiv6",
+                            params: filter
+                        })
+                    );
+                });
+            } else {
+                promises.push(
+                    this.OvhHttp.get(`/hosting/web/${serviceName}/envVar`, {
+                        rootPath: "apiv6"
+                    })
+                );
+            }
+
+            return this.$q.allSettled(promises).then(
```

Comment:
Please use the 
```javascript
.then(...)
.catch((error) => ...)
```

notation instead

Edit : nevermind, `(err) => this.$q.reject(err)` is useless, please remove it